apps/distgo/config/config.go

```diff
@@ -552,25 +564,68 @@ func (cfg *RPMDist) ToParams() params.RPMDistInfo {
 	}
 }
 
+func (cfg *DockerImageInfo) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// unmarshal type alias (uses default unmarshal strategy)
+	type rawImageInfoConfigAlias DockerImageInfo
+	var rawAliasConfig rawImageInfoConfigAlias
+	if err := unmarshal(&rawAliasConfig); err != nil {
+		return err
+	}
+
+	rawImageInfoConfig := DockerImageInfo(rawAliasConfig)
+	switch rawImageInfoConfig.Type {
+	case "sls":
+		type typedRawConfig struct {
+			Type string
+			Data SLSDockerImageInfo
+		}
+		var rawSLS typedRawConfig
+		if err := unmarshal(&rawSLS); err != nil {
+			return err
+		}
+		rawImageInfoConfig.Data = rawSLS.Data
+	}
+	*cfg = rawImageInfoConfig
+	return nil
+}
+
 func (cfg *DockerImage) ToParams() (params.DockerImage, error) {
-	var deps params.DockerDeps
+	var deps []params.DockerDep
 	for _, dep := range cfg.Deps {
 		depType, err := params.ToDockerDepType(dep.Type)
 		if err != nil {
-			return params.DockerImage{}, nil
+			return nil, nil
 		}
 		deps = append(deps, params.DockerDep{
 			Product:    dep.Product,
 			Type:       depType,
 			TargetFile: dep.TargetFile,
 		})
 	}
-	return params.DockerImage{
+	dockerImage := params.DefaultDockerImage{
 		Repository: cfg.Repository,
 		Tag:        cfg.Tag,
 		ContextDir: cfg.ContextDir,
 		Deps:       deps,
-	}, nil
+	}
+	switch cfg.Info.Type {
+	case "sls":
+		slsInfo := SLSDockerImageInfo{}
+		if err := mapstructure.Decode(cfg.Info.Data, &slsInfo); err != nil {
+			return nil, errors.Wrap(err, "Failed to unmarshal the sls image info")
+		}
+		return &params.SLSDockerImage{
+			DefaultDockerImage: dockerImage,
+			ProuductType:       slsInfo.ProuductType,
+			GroupID:            slsInfo.GroupID,
+			Extensions:         slsInfo.Extensions,
+		}, nil
+	case "":
```

Comment:
Currently, the default docker image does not need any extra info. So, chose to return the default docker image in case the type is empty.